### ISSUE TITLE
Modernize command structure

### ISSUE THREAD

**Issue by dchiller** [DDMAL/cantus#405]
Translate commands from old Django cli structure to new one (ie. with argparse-like syntax)

**Comment by eliwaksbaum** [comment]
I'm also curious which commands are even necessary anymore. I have a hunch more than a few are from older workflows and can be deprecated. @napulen thoughts?

**Comment by napulen** [comment]
@eliwaksbaum your hunch is correct! But I don't know which ones, unfortunately. We can all 3 take a look at them during a pair-programming session and decide.

**Comment by dchiller** [comment]
Yeah, now having re-worked some of them, they do seem sort of clunky with arguments, etc. unless the cli functionality is useful...which may be but I suspect that we could just as easily dispatch from the built-in admin pages.

**Comment by dchiller** [comment]
I've done a little review of the different commands we have and categorized them.

### Used in current site + w/ good arg and help structure
- `import_data`
- `import_folio_mapping`
- `refresh_solr`
- `test`

### Maybe useful, but could use re-working
- `flush_chant_data`: Maybe useful to have a flush and reindex command, especially if we want to have a scheduled re-indexer that can just call it. But I think this would need to be re-worked to allow for more fine-tuned use (rather than a complete flush). It should then maybe called in `refresh_solr` rather than the delete function being re-written there. In other words, we could rework so that to completely refresh the solr index you need only two commands: `flush_solr` (or some such) and `index_solr`(which could be a re-working of `refresh_solr`). 

### Commands that may be deprecated
- `reindex_all_data`: Not exactly sure what this is trying to do. Doesn't seem to be called anywhere. Looks like maybe it is taking advantage of the signals to reindex all data in solr? But there are more efficient and straightforward ways to do this, especially with `refresh_solr`.
- `reindex_manuscript`: Seems to replicate `refresh_solr --manuscript_ids ID --all_types`. Not called anywhere.
- `auto_manage_mei_data`, `flush_mei_data`,`import_mei_data`, `remove_mei_data`: Highly particularized to a few specific manuscripts. Called from each other, but nowhere else in codebase.
- `convert_gamera_xml2mei`: Not sure this is doing anything (and not sure what it is supposed to do. Not called anywhere.
- `wait_until_solr_ready`: Seems pretty good, but not used anywhere. Wonder why it was written?


### Still to categorize (just wanted to save)
- `generate_public_datadump`
- `normalize_chant_csv`
- `pick_neume_exemplars`
- `update_public_manuscripts`

**Comment by eliwaksbaum** [comment]
I agree with this categorization! My thoughts on the last four:
- pick_neume_exemplars seems to be a part of the old mei workflow and can go
- normalize_chant_csv can also probably go? doesn't seem like we import data from csvs anymore
- update_public_manuscripts can just be done on the admin site, right
- not sure about generate_public_datadump. could be useful to like save the state of the database, but can't think of a specific use case

**Comment by dchiller** [comment]
Agreed! Just to add:

- `update_public_manuscripts` does sort of allow for a sort of bulk load, but I think the import_data function can also do that? If we want bulk loading by csv then we should just extend the inputs of `import_data`.
- `pick_neume_exemplars`: yeah, looks like this is OMR search -- not clear tbh where that is and whether this would be necessary.

**Comment by napulen** [comment]
Thanks for breaking down the commands @dchiller!

**Comment by dchiller** [comment]
Will do!
